CMakeLists.txt

```diff
@@ -339,6 +342,12 @@ if(ACTS_BUILD_BENCHMARKS OR ACTS_BUILD_INTEGRATIONTESTS OR ACTS_BUILD_UNITTESTS)
   add_subdirectory(Tests)
 endif()
 
+#generate debug information
+if(ACTS_ENABLE_CPU_PROFILING)
+  message(STATUS "Added -g compile flag")
+  add_compile_options(-g)
```

Comment:
I'm a bit concerned about using this to set `-g` globally here. Could this be pushed to `ActsCompilerOptions.cmake` and use `ACTS_CXX_FLAGS` and `CMAKE_CXX_FLAGS` there?

Reply:
I've found that the `-g` option is unnecessary to enable profiling so I've removed it. Thanks for the feedback.

Reply:
I originally thought it was necessary and could've sworn that I'd tried to run the profiler without `-g` to no avail. However, since this was raised I've successfully run the profiler on ACTS and a standalone program without it on two machines and it *seems* fine. I can't find any reference to the flag in gperf's documentation despite it being included in most example articles from other sites I've seen. More than happy to include `-g` again to ensure correct behaviour.

Reply:
Sorry, I don't have time to check this right now, but I seem to remember that the ACTS default was to build with debug with optimisation. If so, then `-g` will be there even if you don't add it explicitly. You can check that by looking at the compiler command in the build log, or in the `ps` output when building.

Reply:
@timadye's right, thanks.